services/motion/motion.go

```diff
@@ -186,8 +186,115 @@ type PlanWithStatus struct {
 }
 
 // A Service controls the flow of moving components.
+//
+// Move example:
+//
+//	// Assumes a gripper with the name "my_gripper" is configured on the machine
+//	gripperName := gripper.Named("my_gripper")
+//
+//	// Get the pose of the gripper
+//	myArmMotionPose, err := motionService.GetPose(context.Background(), gripperName, referenceframe.World, nil, nil)
+//
+//	log.Printf("Position of my_gripper from the motion service: %v", myArmMotionPose.Pose().Point())
+//	log.Printf("Orientation of my_gripper from the motion service: %v", myArmMotionPose.Pose().Orientation())
+//
+// MoveOnMap example:
+//
+//	// Assumes a base with the name "my_base" is configured on the machine
+//	myBaseResourceName := base.Named("my_base")
+//	mySLAMServiceResourceName := slam.Named("my_slam_service")
+//
+//	// Define a destination Pose
+//	myPose := spatialmath.NewPoseFromPoint(r3.Vector{Y: 10})
+//
+//	// Move the base component to the destination pose
+//	executionID, err := motionService.MoveOnMap(context.Background(), motion.MoveOnMapReq{
+//		ComponentName: myBaseResourceName,
+//		Destination:   myPose,
+//		SlamName:      mySLAMServiceResourceName,
+//	})
+//
+//	err = motion.PollHistoryUntilSuccessOrError(
```

Comment:
I would add a comment here saying something like 
`MoveOnMap is a non-blocking method and this line can optionally be added to block until the movement is done`